services/server/cg_server/error.py

```diff
@@ -39,6 +42,7 @@ def wrapper_db_error(*args, **kwargs):
                 return make_response((str(e), 500))
 
             conn.commit()
+            conn_pool.putconn(conn)
```

Comment:
Why is `putconn()` called in both the error wrapper and inside each function? Isn't the connection returned to the pool at this point? What happens if you remove the `putconn()` call from each API function?